akka-http-bench-jmh/src/main/scala/akka/http/impl/model/parser/UriParserBenchmark.scala

```diff
@@ -0,0 +1,34 @@
+package akka.http.impl.model.parser
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri
+import akka.parboiled2.UTF8
+import org.openjdk.jmh.annotations._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+class UriParserBenchmark {
+  implicit val system: ActorSystem = ActorSystem("uri-parser-benchmark")
+
+  val url = "http://any.hostname?param1=111&amp;param2=222"
+
+  @Setup
+  def setup(): Unit = ()
+
+  @TearDown
+  def tearDown(): Unit = {
+    Await.result(system.terminate(), 5.seconds)
+  }
+
+  @Benchmark
+  def bench_parse_uri(): Unit = {
+    Uri(url).query(UTF8, Uri.ParsingMode.Relaxed)
```

Comment:
I think you have to return (or explicitly `Blackhole`) the result to make double-sure the optimizer doesn't remove the thing we're trying to benchmark - though I see we haven't done that in `LineParserBenchmark` either it seems...